From 098c5ac28bbc816ec479cc7b728bda989fe2d03e Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Mon, 26 Nov 2018 12:35:42 +0100
Subject: [PATCH 0221/1239] i2c: mv_i2c: warn if not supported speed mode
 chosen

Hitherto user could pass any speed value, while only two speeds mode were
currently supported. Warn if not supported speed mode is chosen.

Change-Id: I25431b90c1f6ecbecbbab1907291ea38f84448ad
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
---
 drivers/i2c/mv_i2c.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/drivers/i2c/mv_i2c.c b/drivers/i2c/mv_i2c.c
index d1803a6046..227c3f4457 100644
--- a/drivers/i2c/mv_i2c.c
+++ b/drivers/i2c/mv_i2c.c
@@ -567,10 +567,18 @@ static int mv_i2c_set_bus_speed(struct udevice *bus, unsigned int speed)
 	struct mv_i2c_priv *priv = dev_get_priv(bus);
 	u32 val;
 
-	if (speed > 100000)
-		val = ICR_FM;
-	else
+	switch (speed) {
+	case 100000:
 		val = ICR_SM;
+		break;
+	case 400000:
+		val = ICR_FM;
+		break;
+	default:
+		printf("Only 100k and 400k modes supported\n");
+		return -EOPNOTSUPP;
+	}
+
 	clrsetbits_le32(&priv->base->icr, ICR_MODE_MASK, val);
 
 	return 0;
-- 
2.29.0

